### PR TITLE
Various Ninja Buffs (More glove targets, untrackability, grille smashin', wire slashin')

### PR DESCRIPTION
**Pull request by Kurfursten**
🆑 
* rscadd: Ninja gloves now drain anything with a cell (incl tasers, eguns, cyborgs, mechas). If they are on cooldown or the target has less than 100 charge, you will interact with the target normally. Feel free to headpat that unpowered borg.
* rscadd: Wearing ninja gloves renders you untrackable by an AI.
* tweak: The ninja katana no longer conducts electricity. 
* tweak: You may now slice through cables on the ground with any sharpness >=1 item. Hyaa!
* bugfix: Objects will no longer slide toward you if you can't pick them up due to their prepickup effect.
* bugfix: You can only eat pizza roll shuriken, not 3D printed shuriken
* rscadd: You can now smash grilles with throw impacts. No longer will you be able to smash a window but not grilles with shuriken.

### CODE REVIEW

File: code/datums/gamemode/role/ninja.dm [hunk 1]
```diff
@@ -204,20 +192,52 @@
 			if(H.mind.GetRole(NINJA))
 				if(istype(A,/obj/machinery/power/apc))
```
Comment (jknpj) [inline code/datums/gamemode/role/ninja.dm]: kill alll this istype madness and use a single get_cell

**Review by jknpj — CHANGES_REQUESTED** [review]

**Comment by DrSnips** [comment]
Holy unatomic uncles batman!

**Comment by eplgr** [comment]
Do ALL sharp objects work properly with electricity? Or maybe, do we need to balance sharp objects? Does it make sense to get shocked with a screwdriver or not? Etc.

Also, super unatomic PR and the title has almost nothing to do with most of the changes.

**Comment by Kurfursten** [comment]
In this order:
1. Yes - there is no need for any special behavior here
2. No - there is no need for balancing sharp objects (or even if there was, it wouldn't be in scope of this PR), all sharp objects already have a defined sharpness.
3. No - Every item already has its own Siemen's Coefficient. There's no need to discuss the merits of screwdrivers shocking you because they also already have a Siemen's value. It's not in scope to discuss that here.
4. It's all mostly related but I can see how you might say that. The only thing that arguably stands out is the wires.
5. No idea on this one. The title clearly says it's all related to improving ninja quality of life. If you don't see that, I don't think I can make a more clear title. (I will endeavor to improve clarity for you though, best effort)

**Comment by Kurfursten** [comment]
Updated to use get_cell() as requested

Also now grilles can be smashed by throwing shuriken (or anything else) at them. Incidentally I can split these into separate PRs if anyone actually *does* find one of them objectionable enough to have a separate conversation on. I was surprised people cared so much about wirecutting.

**Review by jknpj — APPROVED** [review]

File: code/datums/gamemode/role/ninja.dm [hunk 2]
```diff
@@ -191,33 +180,42 @@
...
-					playsound(APC, pick(lightning_sound), 100, 1, "vary" = 0)
-					APC.charging = 0
-					APC.chargecount = 0
-					cooldown = world.time + 10 SECONDS
+					A.update_icon()
+					return TRUE //Will not perform the normal interaction if drained the cell
 	else
 		..()
 
+/obj/item/clothing/gloves/ninja/proc/draincell(var/obj/item/weapon/cell/C,mob/user)
```
Comment (jknpj) [inline code/datums/gamemode/role/ninja.dm:210]: Maybe a C check? get_cell might give you null

**Comment by jknpj** [comment]
Needs to update the :cl: because now they can drain anything that has a internal cell

**Comment by MadmanMartian** [comment]
Hm, maybe the tracking-blocking should be relocated to a helmet, not the gloves?

**Comment by SonixApache** [comment]
>bugfix: Objects will no longer slide toward you if you can't pick them up.

undo this

**Comment by Kurfursten** [comment]
I updated the changelog as requested and I'll move object sliding to its own bugfix PR

I must remember to convert the hologram projector to use trange for turfs as this is much less costly.

**Comment by Kurfursten** [comment]
@SonixApache I realize now why you care about that change. I should've been more clear in my changelog, this only stops items from sliding toward you if you would fail its prepickup. This does not affect, say, sliding items while crawling. (Nor does it change the current behavior regarding trying to pick up an item without that hand, for example)

Do you still care?

**Comment by SonixApache** [comment]
i no longer care

**Pull request merged**